platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -2508,6 +2508,10 @@ void DisplayServerX11::window_set_ime_active(const bool p_active, WindowID p_win
 		return;
 	}
 
+	if (p_active == wd.ime_active) {
+		return;
+	}
```

Comment:
This won't work either, IME can get closed event if it's not explicitly deactivated, and input controls call this on every change, to reactivate it.

Reply:
I guess it should check if the IME window is focused (`XGetInputFocus`?) instead of checking `wd.ime_active`.